src/main/java/net/syscon/elite/service/CourtHearingsService.java

```diff
@@ -45,33 +50,65 @@ public CourtHearingsService(final OffenderBookingRepository offenderBookingRepos
 
     @Transactional
     public CourtHearing scheduleHearing(final Long bookingId, final PrisonToCourtHearing hearing) {
-        // TODO throw entity not found exception
-        var offenderBooking = offenderBookingRepository.findById(bookingId).orElseThrow();
-
-        // TODO throw entity not found exception (check is active)
-        var courtCase = offenderBooking.getCourtCaseBy(hearing.getCourtCaseId()).orElseThrow();
+        final var offenderBooking = getActiveOffenderBookingFor(bookingId);
 
-        // TODO throw entity not found exception (check is prison?)
-        var locationOfPrison = agencyLocationRepository.findById(hearing.getFromPrisonLocation()).orElseThrow();
+        final var courtCase = getActiveCourtCaseFor(hearing.getCourtCaseId(), offenderBooking);
 
-        // TODO throw entity not found exception (check is court)
-        var locationOfCourt = agencyLocationRepository.findById(hearing.getToCourtLocation()).orElseThrow();
+        checkPrisonLocationSameAsOffenderBooking(hearing.getFromPrisonLocation(), offenderBooking);
 
-        // TODO sort out reference codes!!!
         CourtEvent courtEvent = CourtEvent.builder()
+                .courtLocation(getCourtFor(hearing.getToCourtLocation()))
                 .courtEventType(eventTypeRepository.findById(EventType.COURT).orElseThrow())
-                .courtLocation(locationOfCourt)
                 .directionCode("OUT")
                 .eventDate(hearing.getCourtHearingDateTime().toLocalDate())
                 .eventStatus(eventStatusRepository.findById(EventStatus.SCHEDULED).orElseThrow())
-                .offenderCourtCase(courtCase)
                 .offenderBooking(offenderBooking)
+                .offenderCourtCase(courtCase)
                 .startTime(hearing.getCourtHearingDateTime())
                 .build();
 
         return toCourtHearing(courtEventRepository.save(courtEvent));
     }
 
+    private OffenderBooking getActiveOffenderBookingFor(final Long bookingId) {
+        final var offenderBooking = offenderBookingRepository.findById(bookingId).orElseThrow(EntityNotFoundException.withId(bookingId));
+
+        if (offenderBooking.isActive()) {
+            return offenderBooking;
+        }
+
+        throw new IllegalArgumentException(format("Offender booking with id [%d] is not active.", bookingId));
+    }
+
+    private OffenderCourtCase getActiveCourtCaseFor(final Long caseId, final OffenderBooking offenderBooking) {
+        final var courtCase = offenderBooking.getCourtCaseBy(caseId).orElseThrow(EntityNotFoundException.withId(caseId));
+
+        if (courtCase.isActive()) {
+            return courtCase;
+        }
+
+        throw new IllegalArgumentException(format("Court case with id [%d] is not active.", caseId));
```

Comment:
these could potentially all be replaced by `checkState`s?

Reply:
apache commons `Validate` is used quite a bit too.

Reply:
Done.  Updated to use Preconditions.